src/main/java/Doemon.java

```diff
@@ -0,0 +1,189 @@
+            "                     \\              /  /\n" +
+            "                       |== (t) ===|____";
+
+    /**
+     * Introduction string that is printed when Doemon is started.
+     */
+    private static String introStr = "Hello I'm\n" + logo + "\t\t\tDoemon!";
+
+    /**
+     * The string that is printed when Doemon is exited.
+     */
+    private static String exitStr = "I'm going to sleep now...See you again soon!";
+
+    /**
+     * A fixed-sized string array that stores user input.
+     */
+    private ArrayList<Task> tasks = new ArrayList<>();
+
+    public static void main(String[] args) {
+        // Introduction
+        System.out.println(output(introStr));
+
+        new Doemon().start();
+    }
+
+    /**
+     * Starts the Doemon chatbot.
+     */
+    private void start() {
+        // Handling inputs
+        Scanner sc = new Scanner(System.in);
+
+        while (true) {
+            String inputStr = sc.nextLine();
+
+            // Exit the chatbot
+            if (inputStr.equals("bye")) {
+                System.out.println(output(exitStr));
+                break;
+            }
+
+            // List tasks
+            if (inputStr.equals("list")) {
+                int listNum = 1;
+                String listStr = "Here is what's on my bread:\n\t";
+                for (Task task : tasks) {
+                    if (task == null) break;
+                    listStr += listNum++ + "." + task.toString() + "\n\t";
+                }
+                listStr = listNum == 1 ? "You have no tasks!" : listStr.trim();
+                System.out.println(output(listStr));
+                continue;
+            }
+
+            String[] inputArr = inputStr.split(" ");
+
+            // Check for mark/unmark/delete
+            try {
+                if (hasMarkedOrDeleted(inputArr)) continue;
+            } catch (InvalidTaskNumberException itne) {
+                System.out.println(output(itne.toString()));
+            }
+
+            // Add task
+            try {
+                addTask(inputStr, inputArr);
+            } catch (EmptyTaskException ete) {
+                System.out.println(output(ete.toString()));
+            } catch (InvalidTaskException ite) {
+                System.out.println(output(ite.toString()));
+            } catch (MissingArgumentException mae) {
+                System.out.println(output(mae.toString()));
+            }
+        }
+    }
+
+    /**
+     * Attempts to mark/unmark/delete a task depending on the input. Returns a boolean indicating if the command was
+     * to mark/unmark/delete.
+     * @param inputArr the user-inputted string split into String array using a space delimiter
+     * @return a boolean indicating if the command was a mark/unmark/delete command
+     * @throws InvalidTaskNumberException if the task number indicated does not exist
+     */
+    private boolean hasMarkedOrDeleted(String[] inputArr) throws InvalidTaskNumberException{
```

Comment:
Perhaps it's good to shorten the method length by creating separate methods to handle the `System.out.println` and parsing commands? So that it can follow this coding standard:

> Avoid Long Methods
  Can improve code quality using technique: avoid long methods
Be wary when a method is longer than the computer screen, and take corrective action when it goes beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle.